src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/Telemetry/TelemetryReporter.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common.Telemetry
+{
+    internal static class TelemetryReporter
+    {
+        private static readonly object _guard = new();
+        private static ImmutableArray<TelemetrySession> s_telemetrySessions = ImmutableArray<TelemetrySession>.Empty;
+
+        public static void RegisterTelemetrySesssion(TelemetrySession session)
```

Comment:
Technically we can support multiple sessions and report the same telemetry on each. It's not a requirement, and again this was copied from Roslyn. Most callers won't care about this